### ISSUE TITLE
Agent.prototype.isMute()

### ISSUE THREAD

**Issue by mfulton26** [amazon-connect/amazon-connect-streams#917]
There's `Agent.prototype.mute()` and `Agent.prototype.unmute()` but no `Agent.prototype.isMute()`.

Can we get an `isMute()` (or `isMuted()`) method on `Agent` to query if the local media stream is muted or not?

**Comment by andywang219** [comment]
Hi!

As a workaround for now, you could do the following:

```
// get the agent's connection
const agentConnection = contact.getAgentConnection();

// use voiceConnection.isMute()
if (agentConnection.isMute()) {
...
}
```
[voiceConnection.isMute()](https://github.com/amazon-connect/amazon-connect-streams/blob/master/Documentation.md#voiceconnectionismute)

**Comment by mfulton26** [comment]
> voiceConnection.isMute()

that checks if the connection is muted server side; I want to check to see if it is muted client side (which is what `Agent.prototype.mute()` does)

**Comment by mfulton26** [comment]
> voiceConnection.isMute()

@andywang219, I don't think that works, it checks if the connection is muted server side; I want to check to see if it is muted client side (which is what `Agent.prototype.mute()` does)

**Comment by andywang219** [comment]
We've actually started using the value that is returned in `voiceConnection.isMute()` to mute the agent's local media stream now. This allows us to maintain the mute state for the agent across all instances (ie. across different browsers).

One other workaround I can think of is calling `getUserMedia()` to grab all available local media streams for the agent, look for the media stream that is active, and then checking to see if its muted.

**Comment by mfulton26** [comment]
using `voiceConnect.isMute()` sounds appropriate then (except I can't query it unless I'm on an actual call… I'm guessing that doesn't matter though if new calls start as unmuted)

I wasn't aware that `getUserMedia()` can be used to list active streams… is that a thing? also, if the media stream is active in the CCP iframe then I don't think my app would have access to it… I'm not sure though, I haven't tested it

**Comment by mfulton26** [comment]
> It has been a while and we are closing this for now. Please reopen if this is still an issue

I do not have permissions to reopen.
